Lorawan regression in v2.7rc4 for US915 and AU915 regions

**Describe the bug**

The lorawan sample fails to link if the region setting in the sample configuration is changed from the default.

IN868, EU868, AS923 work

US915 AU915 fail

**To Reproduce**

Steps to reproduce the behavior:

1. cd zephyrproject/zephyr
2. perl -pi -e 's/LORAMAC_REGION_[A-Z0-9]+=y/LORAMAC_REGION_AU915=y/' samples/subsys/lorawan/class_a/prj.conf
3.  west build -p -b 96b_wistrio  samples/subsys/lorawan/class_a (fails)

```
/Users/chris/Applications/gcc-arm-none-eabi-8-2019-q3-update/bin/../lib/gcc/arm-none-eabi/8.3.1/../../../../arm-none-eabi/bin/ld.bfd: zephyr/subsys/lorawan/libloramac-node.a(RegionUS915.c.obj): in function `RegionUS915GetPhyParam':
/Users/chris/zephyrproject-main/modules/lib/loramac-node/src/mac/region/RegionUS915.c:254: undefined reference to `RegionBaseUSCalcDownlinkFrequency'
/Users/chris/Applications/gcc-arm-none-eabi-8-2019-q3-update/bin/../lib/gcc/arm-none-eabi/8.3.1/../../../../arm-none-eabi/bin/ld.bfd: zephyr/subsys/lorawan/libloramac-node.a(RegionUS915.c.obj): in function `RegionUS915NextChannel':
/Users/chris/zephyrproject-main/modules/lib/loramac-node/src/mac/region/RegionUS915.c:868: undefined reference to `RegionBaseUSComputeNext125kHzJoinChannel'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /Users/chris/Microcode/zephyr/zephyrproject-main/zephyr/build
```

**Expected behavior**

The same sample builds successfully in v2.6-branch

**Impact**

Cannot use LoRaWAN on this release

**Logs and console output**

```
zephyr on  v2.7-branch took 1m4s
 1% ❯ perl -pi -e 's/LORAMAC_REGION_[A-Z0-9]+=y/LORAMAC_REGION_US915=y/' samples/subsys/lorawan/class_a/prj.conf

zephyr on  v2.7-branch [!?]
 10% ❯ west build -p -b 96b_wistrio  samples/subsys/lorawan/class_a
-- west build: making build dir /Users/chris/Microcode/zephyr/zephyrproject-main/zephyr/build pristine
-- west build: generating a build system
Including boilerplate (Zephyr base): /Users/chris/zephyrproject-main/zephyr/cmake/app/boilerplate.cmake
-- Application: /Users/chris/zephyrproject-main/zephyr/samples/subsys/lorawan/class_a
-- Zephyr version: 2.7.0-rc4 (/Users/chris/zephyrproject-main/zephyr), build: v2.7.0-rc4-38-g8a97c83040c0
-- Found Python3: /usr/local/opt/python@3.9/bin/python3.9 (found suitable exact version "3.9.7") found components: Interpreter
-- Found west (found suitable version "0.11.0", minimum required is "0.7.1")
-- Board: 96b_wistrio
-- Cache files will be written to: /Users/chris/Library/Caches/zephyr
-- Found dtc: /usr/local/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found toolchain: gnuarmemb (/Users/chris/Applications/gnuarmemb)
-- Found BOARD.dts: /Users/chris/zephyrproject-main/zephyr/boards/arm/96b_wistrio/96b_wistrio.dts
-- Generated zephyr.dts: /Users/chris/zephyrproject-main/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /Users/chris/zephyrproject-main/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /Users/chris/zephyrproject-main/zephyr/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /Users/chris/zephyrproject-main/zephyr/build/zephyr/dts.cmake
Parsing /Users/chris/zephyrproject-main/zephyr/Kconfig
Loaded configuration '/Users/chris/zephyrproject-main/zephyr/boards/arm/96b_wistrio/96b_wistrio_defconfig'
Merged configuration '/Users/chris/zephyrproject-main/zephyr/samples/subsys/lorawan/class_a/prj.conf'
Configuration saved to '/Users/chris/zephyrproject-main/zephyr/build/zephyr/.config'
Kconfig header saved to '/Users/chris/zephyrproject-main/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 8.3.1
-- The CXX compiler identification is GNU 8.3.1
-- The ASM compiler identification is GNU
-- Found assembler: /Users/chris/Applications/gnuarmemb/bin/arm-none-eabi-gcc
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/chris/zephyrproject-main/zephyr/build
-- west build: building application
[1/169] Preparing syscall dependency handling

[162/169] Linking C executable zephyr/zephyr_prebuilt.elf
FAILED: zephyr/zephyr_prebuilt.elf zephyr/zephyr_prebuilt.map /Users/chris/zephyrproject-main/zephyr/build/zephyr/zephyr_prebuilt.map
: && ccache /Users/chris/Applications/gnuarmemb/bin/arm-none-eabi-gcc   zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj -o zephyr/zephyr_prebuilt.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_prebuilt.cm
d  -Wl,-Map=/Users/chris/zephyrproject-main/zephyr/build/zephyr/zephyr_prebuilt.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/
libarch__arm__core__aarch32__cortex_m.a  zephyr/arch/arch/arm/core/aarch32/mpu/libarch__arm__core__aarch32__mpu.a  zephyr/lib/libc/newlib/liblib__libc__newlib.a  zephyr/lib/posix/liblib__posix.a  zephyr/soc/arm/common/cortex_m/libsoc__arm__common__cortex_m.a  zephyr/boards/arm/96b_wist
rio/libboards__arm__96b_wistrio.a  zephyr/subsys/lorawan/libloramac-node.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gp
io.a  zephyr/drivers/spi/libdrivers__spi.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/stm32/stm32cube/lib..__modules__hal__stm32__stm32cube.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/Users/chris/Applications/gcc-arm-non
e-eabi-8-2019-q3-update/bin/../lib/gcc/arm-none-eabi/8.3.1/thumb/v7-m/nofp"  -L/Users/chris/zephyrproject-main/zephyr/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -mcpu=cortex-m3  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-c
ommon=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -no-pie  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -lm  -Wl,-lc  -L"/Users/chris/Applications/gnuarmemb/arm-none-eabi"/lib/thumb/v7-m/nofp  -Wl,-lgcc  -lc  -specs=nano.sp
ecs && cd /Users/chris/zephyrproject-main/zephyr/build/zephyr && /usr/local/Cellar/cmake/3.21.2/bin/cmake -E echo
/Users/chris/Applications/gcc-arm-none-eabi-8-2019-q3-update/bin/../lib/gcc/arm-none-eabi/8.3.1/../../../../arm-none-eabi/bin/ld.bfd: zephyr/subsys/lorawan/libloramac-node.a(RegionUS915.c.obj): in function `RegionUS915GetPhyParam':
/Users/chris/zephyrproject-main/modules/lib/loramac-node/src/mac/region/RegionUS915.c:254: undefined reference to `RegionBaseUSCalcDownlinkFrequency'
/Users/chris/Applications/gcc-arm-none-eabi-8-2019-q3-update/bin/../lib/gcc/arm-none-eabi/8.3.1/../../../../arm-none-eabi/bin/ld.bfd: zephyr/subsys/lorawan/libloramac-node.a(RegionUS915.c.obj): in function `RegionUS915NextChannel':
/Users/chris/zephyrproject-main/modules/lib/loramac-node/src/mac/region/RegionUS915.c:868: undefined reference to `RegionBaseUSComputeNext125kHzJoinChannel'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /Users/chris/Microcode/zephyr/zephyrproject-main/zephyr/build
```

**Environment (please complete the following information):**

On MacOS, using various ARM boards (including  96b_wistrio as that's what the sample assumes).

8a97c83040 - (3 days ago) arm: aarch32: mpu: Fix build issue with assert — Kumar Gala (HEAD -> v2.7-branch, origin/v2.7-branch)

**Additional context**

I'll attempt to diagnose further, but I wanted to get this report in ASAP given its a regression in a release candidate.

--- a/modules/loramac-node/CMakeLists.txt
+++ b/modules/loramac-node/CMakeLists.txt
@@ -61,6 +61,7 @@ zephyr_library_sources_ifdef(CONFIG_LORAMAC_REGION_EU868
   ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionEU868.c
 )
 zephyr_library_sources_ifdef(CONFIG_LORAMAC_REGION_US915
+  ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionBaseUS.c
   ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionUS915.c
 )
 zephyr_library_sources_ifdef(CONFIG_LORAMAC_REGION_CN779
@@ -70,6 +71,7 @@ zephyr_library_sources_ifdef(CONFIG_LORAMAC_REGION_EU433
   ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionEU433.c
 )
 zephyr_library_sources_ifdef(CONFIG_LORAMAC_REGION_AU915
+  ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionBaseUS.c
   ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionAU915.c
 )
 zephyr_library_sources_ifdef(CONFIG_LORAMAC_REGION_AS923

